### DISCUSSION TITLE
Getting started with spark nlp and facing errors in setInputCols

### DISCUSSION THREAD

**Discussion by AdityaDesai17** [JohnSnowLabs/spark-nlp#13131] [General]
https://nlp.johnsnowlabs.com/2020/12/29/ner_aspect_based_sentiment_en.html

<img width="1454" alt="image" src="https://user-images.githubusercontent.com/63996781/203564538-ae62eed5-e2fb-422f-94a9-328e319a7fd2.png">

https://nlp.johnsnowlabs.com/2021/11/22/sentiment_vivekn_en.html

<img width="1454" alt="image" src="https://user-images.githubusercontent.com/63996781/203565541-dfd98a8f-0332-4119-bf9d-bd39303971da.png">

Hey guys, I need assistance with getting started with this. I am trying to implement sentimental analysis for one of my projects and was referring to the documentation. I executed two of the models directly from their colab implementation, but both of them ran into the same issue. I have attached the screenshots as well. Can someone assist me with what can be the error? Am I missing something important here? please lmk. I am also adding the links to the original code I am using as a reference.

**Comment by maziyarpanahi** [comment]
Hi,

Could you please share your full code? The errors are related to code that is not in the links you shared.

**Reply by maziyarpanahi** [reply]
Sorry @AdityaDesai17, found the Colab link here and I am going to have a look: https://colab.research.google.com/github/JohnSnowLabs/spark-nlp-workshop/blob/master/tutorials/streamlit_notebooks/ABSA_Inference.ipynb#scrollTo=LWoA7SChHu41

In Spark NLP 4.2.3 release we started restricting the provided number of columns in Python to be similar to Scala. If an annotator expects 2 in previous versions we could have passed 3 columns. We fixed that and unfortunately, there are some examples/demos which were not made carefully so we are going to fix them when they are reported.

I will come back to you regarding your errors. @danilojsl

**Reply by maziyarpanahi** [reply]
@AdityaDesai17 can confirm, it's a bug and we will fix it in our next release. Many thanks for reporting these 2 annotators. (https://github.com/JohnSnowLabs/spark-nlp/issues/13133)

For the time being, you can install `pip install spark-nlp==4.2.2 pyspark` instead and you should be fine until we release `4.2.4` to fix this.
